hack/tools/release/notes/mock_github.go

```diff
@@ -0,0 +1,158 @@
+//go:build tools
```

Comment:
This should be named maybe `github_test.go` so its not usable by accident for prod code.